mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -1202,21 +1205,28 @@ static class WorkerNumberGenerator {
         }
 
         private void advance(MantisJobMetadataImpl mantisJobMetaData, MantisJobStore jobStore) {
+            currLimit += incrementStep;
             try {
-                currLimit += incrementStep;
-
                 mantisJobMetaData.setNextWorkerNumberToUse(currLimit, jobStore);
+                pendingFlushCnt.set(0);
             } catch (Exception e) {
+                if (pendingFlushCnt.incrementAndGet() > MAX_PENDING_FLUSH_CNT) {
```

Comment:
the write failure will still throw the runtime exception?

Reply:
oops, missed an early return here during a refactoring I did. Thanks!